### PR TITLE
[Feature] add monoflex benchmark

### PR DESCRIPTION
**Pull request by ZCMax**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification
Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by ZCMax** [comment]
Notice: The training performance is not stable.

File: configs/monoflex/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# SMOKE: Single-Stage Monocular 3D Object Detection via Keypoint Estimation
```
Comment (Tai-Wang) [inline configs/monoflex/README.md:1]: Fix this title

File: configs/monoflex/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+<!-- [ABSTRACT] -->
+
+The precise localization of 3D objects from a single image without depth information is a highly challenging problem. Most existing methods adopt the same approach for
+all objects regardless of their diverse distributions, leading
+to limited performance for truncated objects. In this paper, we propose a flexible framework for monocular 3D object detection which explicitly decouples the truncated objects and adaptively combines multiple approaches for object depth estimation. Specifically, we decouple the edge
+of the feature map for predicting long-tail truncated objects so that the optimization of normal objects is not influenced. Furthermore, we formulate the object depth estimation as an uncertainty-guided ensemble of directly regressed
+object depth and solved depths from different groups of
+keypoints. Experiments demonstrate that our method outperforms the state-of-the-art method by relatively 27% for
+the moderate level and 30% for the hard level in the test
+set of KITTI benchmark while maintaining real-time efficiency.
```
Comment (Tai-Wang) [inline configs/monoflex/README.md:14]: Reorganize the abstract (make it as a single paragraph)

File: configs/monoflex/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    year      = {2021},
+    pages     = {3289-3298}
+}
+```
+
+## Results
+
+### KITTI
+
+|  Backbone   | Lr schd | Mem (GB) | Inf time (fps) | mAP | Download |
```
Comment (Tai-Wang) [inline configs/monoflex/README.md:46]: Refer to the benchmark format in the PGD readme

File: configs/monoflex/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    author    = {Zhang, Yunpeng and Lu, Jiwen and Zhou, Jie},
+    title     = {Objects Are Different: Flexible Monocular 3D Object Detection},
+    booktitle = {Proceedings of the IEEE/CVF Conference on Computer Vision and Pattern Recognition (CVPR)},
+    month     = {June},
+    year      = {2021},
+    pages     = {3289-3298}
+}
+```
+
+## Results
```
Comment (Tai-Wang) [inline configs/monoflex/README.md:35]: Can be adjusted as the new format of README (#1195 )

File: configs/monoflex/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+Note: mAP represents Car moderate 3D strict AP11 results.
+Detailed performance on KITTI 3D detection (3D/BEV) is as follows, evaluated by AP11 and AP40 metric:
+
+|             |     Easy      |    Moderate    |     Hard      |
+|-------------|:-------------:|:--------------:|:-------------:|
+| Car (AP11)  | 28.02 / 36.11 | 21.86 / 29.46  | 19.01 / 24.83 |
+| Car (AP40)  | 23.22 / 32.74 | 17.18 / 24.02  | 15.13 / 20.67 |
+
+Note: mAP represents Car moderate 3D strict AP11 / AP40 results. Because of the limited data for pedestrians and cyclists, the detection performance for these two classes is usually unstable. Therefore, we only list car detection results here. In addition, AP40 is a more recommended metric for reference due to its much better stability. Besides, the training of MonoFlex is unstable, the final result may fluctuate ~1 AP.
```
Comment (Tai-Wang) [inline configs/monoflex/README.md:51]: In addition and Besides are repetitive. May reorganize the expression

**Review by Tai-Wang — COMMENTED** [review]
Just need to revise the notes

**Pull request merged**